### PR TITLE
Add support for remaining Splunk Write options for GCS-to-Splunk template

### PR DESCRIPTION
**Pull request by andreigurau**
Add the missing Splunk options, found in the Pubsub-to-Splunk template, but pulling the options from the Beam SDK

### CODE REVIEW

File: v2/googlecloud-to-splunk/src/main/java/com/google/cloud/teleport/v2/transforms/SplunkConverters.java [hunk 1]
```diff
@@ -134,6 +135,95 @@ public interface SplunkOptions extends PipelineOptions {
     Integer getParallelism();
 
     void setParallelism(Integer parallelism);
+
+    @TemplateParameter.Enum(
+        order = 6,
+        enumOptions = {"PLAINTEXT", "KMS", "SECRET_MANAGER"},
```
Comment (bvolpato) [inline v2/googlecloud-to-splunk/src/main/java/com/google/cloud/teleport/v2/transforms/SplunkConverters.java:141]: Can we also make this `optional = true`?

**Review by bvolpato — CHANGES_REQUESTED** [review]
It looks good, just one concern about a non-optional parameter that could break existing usage

**Comment by andreigurau** [comment]
> It looks good, just one concern about a non-optional parameter that could break existing usage

This template isn't available for public use yet (waiting on the Splunk resource manager to be created first, and then need to write integration tests). That parameter in Pubsub-to-Splunk is required due to that reason you mentioned, but since no one is using this template, I would prefer to keep this parameter mandatory in this case

**Review by bvolpato — DISMISSED** [review]
My bad, LGTM then! Thanks Andrei

**Pull request merged**